src/eradiate/scenes/illumination/_spot.py

```diff
@@ -0,0 +1,227 @@
+        "Can be initialised with a dictionary processed by "
+        ":meth:`.SpectrumFactory.convert`.",
+        type=":class:`~eradiate.scenes.spectra.Spectrum`",
+        init_type=":class:`~eradiate.scenes.spectra.Spectrum` or dict or float",
+        default="1.0 ucc[intensity]",
+    )
+
+    beam_profile: t.Optional[Path] = documented(
+        attrs.field(
+            default=None,
+            converter=attrs.converters.optional(Path),
+        ),
+        type="Path",
+        init_type="path-like",
+        default=None,
+        doc="Path to the file describing the beam profile. Must be a valid bitmap image file.",
+    )
+
+    @classmethod
+    def from_size_at_target(
+        cls,
+        target: np.typing.ArrayLike,
+        direction: np.typing.ArrayLike,
+        spot_radius: pint.Quantity,
+        beam_width: pint.Quantity,
+        **kwargs,
+    ) -> SpotIllumination:
+        """
+        Create a :class:`.SpotIllumination` which
+        illuminates a region of a specified size around the
+        target point.
+
+        The illuminated area is a circle defined by the
+        spot radius around the target point in the plane
+        orthogonal to the spot direction.
+        The spot origin is inferred from the spot radius
+        and a beam width value (in angle units): the
+        spot origin will be positioned closer to the target
+        point as the beam width increases and further from the
+        target point as the beam width decreases.
+
+        Parameters:
+        -----------
+
+        target : array-like
+            Target point for the spot. Unitless values are converted to ``ucc['length']``
+
+        direction : array-like
+            Pointing direction for the spot, towards target.
+
+        spot_radius : float or quantity
+            Radius of the desired spot at the target position.
+            Unitless values are converted to ``ucc['length']``
+
+        beam_width : float or quantity
+            Divergence angle of the spot.
+            Unitless values are converted to ``ucc['angle']``
+
+        **kwargs
+            Remaining keyword arguments are forwarded to the
+            :class:`.SpotIllumination` constructor.
+
+        Returns
+        -------
+        SpotIllumination
+        """
+
+        if "origin" in kwargs:
+            warnings.warn(
+                "The from_size_at_target constructor computes the origin position. "
+                "The specified value will be overwritten."
+            )
+            kwargs.pop("origin")
+
+        target = pinttr.util.ensure_units(target, default_units=ucc.get("length"))
+        half_angle = (
+            pinttr.util.ensure_units(beam_width, default_units=ureg.radian) / 2.0
+        )
+        tan_divergence = np.tan(half_angle)
+        distance = spot_radius / tan_divergence
+
+        origin = target - (direction * distance)
+
+        _, up = mi.coordinate_system(direction)
```

Comment:
See previous comment about the default `up` value: this should be moved to the constructor.